src/app/services/xml-parsers/bibliography-parser.service.ts

```diff
@@ -0,0 +1,96 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { forkJoin, Observable } from 'rxjs';
+import { map, shareReplay } from 'rxjs/operators';
+import { AppConfig } from 'src/app/app.config';
+
+function getStringFromTags(tagLabel: string, cit: Element) {
+  let res = '';
+  Array.from(cit.getElementsByTagName(tagLabel)).map(element => res += element.textContent.replace(/\s+/g, ' ') + ' ');
+
+  return res.trim();
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class BibliographyParserService {
+  private editionUrls = AppConfig.evtSettings.files.editionUrls || [];
+  private findedCits = { complete: 0, total: 0 };
+
+  constructor(
+    private http: HttpClient,
+  ) {
+  }
+
+  public getBibliographicCitations(): Observable<CitationsFeature> {
+    const parser = new DOMParser();
+    const bibliographicCitations: BibliographicCitation[] = [];
+
+    return forkJoin(this.editionUrls.map((path) => this.http.get(path, { responseType: 'text' }))).pipe(
```

Comment:
Why do you make another http call? The xml file has already been recovered and ready to use. Look at the code better.

Reply:
Great, many thanks! I had looked for it, but I hadn't found it before!